### ISSUE TITLE
could not connect to thrift client to port 

### ISSUE THREAD

**Issue by Wannabeperfect** [p4lang/behavioral-model#596]
I'm doing an experiment with multiple queues.The topo is shown as below. I need to get the queue lengths of the four egress ports 5, 6, 7, and 8 of s1. I've made the following changes in simple_switch.cpp according to the method  @antoninbas gave me.
![38780788-907e563e-410e-11e8-8888-2a496e73e388](https://user-images.githubusercontent.com/38399275/39476027-1350f00c-4d4a-11e8-965c-7211b406d489.png)
The simple_switch.cpp is like this:
![2018-05-01 22 18 31](https://user-images.githubusercontent.com/38399275/39476171-91355260-4d4a-11e8-826a-e458d06cef73.png)
And I changed v1model.p4:
![2](https://user-images.githubusercontent.com/38399275/39476242-c8de6eb8-4d4a-11e8-9250-c32164fb5df2.png)
My code is like this,I record the queue length of the four egress ports in registers
![1](https://user-images.githubusercontent.com/38399275/39476791-cd074116-4d4c-11e8-9348-f5e85245470a.png)
![1](https://user-images.githubusercontent.com/38399275/39477126-d35d7f52-4d4d-11e8-9be8-cbf8d73f1ee2.png)
I successfully started the mininet：
![4](https://user-images.githubusercontent.com/38399275/39477766-ccba644c-4d4f-11e8-8cf7-58ecff31cb8c.png)
However,when I want to add entries to the tables,I encounter errors like this:
![1240712483](https://user-images.githubusercontent.com/38399275/39477866-1ac32f5c-4d50-11e8-80da-013cf2f5a296.jpg)
I use ps aux|grep bmv2 to test whether bmv2 run successfully,the result shows as below:
![7](https://user-images.githubusercontent.com/38399275/39477948-5c8ce3a6-4d50-11e8-91bd-44b0345d3b47.png)

It seems like it doesn't run successfully.

So where is the problem?

I only did make clean && make to bmv2, but did not perform such operations on p4c. Is the problem here? Are there any other mistakes?

Thanks ：)

**Comment by antoninbas** [comment]
The switch is crashing. `max_port` is 256 by default, so unless you add a check for the field existence with `has_field`, you need to define all 256 metadata fields, one for each port. Also, you modified v1model.p4 to introduce fields `deq_qdepth_*`, but in the C++ code you try to set `enq_qdepth_*`.

Once you fix these, if the simple_switch process is still crashing, try running it under gdb outside of Mininet:
```
libtool --mode=execute gdb --args ./simple_switch -i 0@veth0 --log-console
```
Then inject a packet on veth0/veth1 to see if the process crashes (you can do it with the scapy CLI for example).

**Comment by Wannabeperfect** [comment]
@antoninbas 
Thanks for your reply.
What I need to measure is the queue length of the four egress ports of s1. Due to my negligence causing a problem in the cpp file, I have now changed it to deq_qdepth_*.
You say max_port is 256 by default. I need to define 256 metadata. For my topology, s1 has only 8 ports, so I changed it to 8 in the cpp file. I performed make clean&&make on bmv2. , But when running ./run_demo.sh, it was found that the switch still failed to start.
![2018-05-02 09 33 48](https://user-images.githubusercontent.com/38399275/39501187-e80e838c-4da8-11e8-8b53-0b1ff387a085.png)
